### ISSUE TITLE
Dir.glob returns doubles, if used with recursive directory matchers "**"

### ISSUE THREAD

**Issue by MartinKoerner** [jruby/jruby#4353]
### Environment

* Mac OS Sierra 10.12.1
* Directory structure with multiple subdirectories:
```
   tmp
    |\2
       |\abc
       |   \2 - file1.pdf
       |   \4 - file2.pdf
       |\def
       |   \2 - file3.pdf
       |   \4 - file4.pdf
```
### Expected Behavior
```ruby
   # MRI
   # rbenv shell 2.3.1
   # irb
      Dir.glob("tmp").size # = 1
      Dir.glob("tmp/2/abc/2/*.pdf").size # = 1
      Dir.glob("tmp/**/**/**/*.pdf").size # = 4
```

### Actual Behavior
```ruby
   # rbenv shell jruby-9.1.6.0
   # jirb
     Dir.glob("tmp").size # = 1
     Dir.glob("tmp/2/abc/2/*.pdf").size # = 1
     Dir.glob("tmp/**/**/**/*.pdf").size # = 12 <- WRONG!
     Dir.glob("tmp/**/**/**/*.pdf").uniq.size # 4 # doubles are equals according to String
```

**Comment by sumitmah** [comment]
@kares does the following PR fixes the issue?
 https://github.com/jruby/jruby/pull/4428

**Comment by kares** [comment]
@sumitmah well, we can't tell for sure until we do not have a test-case, right? :)

**Comment by sumitmah** [comment]
@kares I can add test case, I was concerned about other scenarios? do we have tests for them?

**Comment by headius** [comment]
The PR seems logical. You can run other test cases by running spec/mspec/bin/mspec ci spec/ruby/core/dir or some tests somewhere in test/mri (probably test/mri/ruby/test_dir.rb). You can also add cases to spec/ruby or test/jruby (rather not add to test/mri) that mimic the above cases.

**Comment by sumitmah** [comment]
Thanks! @headius I'll check the specs.

**Comment by enebo** [comment]
@sumitmah I do not want to land this right before 9.1.7.0 but we will merge afterwards.

**Comment by sumitmah** [comment]
@enebo when is 9.1.7.0  release planned?

**Comment by sumitmah** [comment]
@headius @kares I've added test case

**Comment by kares** [comment]
should be resolved as #4428 got merged.
